packages/bpk-component-calendar/src/Week.js

```diff
@@ -87,6 +102,30 @@ class Week extends Component {
     return false;
   }
 
+  componentWillUnmount() {
+    document.removeEventListener('mouseup', this.onMouseUp);
+  }
+
+  onMouseClick(date) {
+    this.props.onDateClick(date);
+    this.setState({ savedDate: null });
+  }
+
+  onMouseDown(date, month) {
+    if (isSameMonth(date, month)) {
+      this.props.onDateMouseDown(null, { date });
+    } else {
+      this.setState({ savedDate: date });
+    }
+  }
+
+  onMouseUp(event) {
+    if (this.state.savedDate != null) {
+      this.props.onDateMouseDown(event, { date: this.state.savedDate });
```

Comment:
Calling a "mouse down" event handler on a mouse up event is surprising.

Reply:
I agree - the 'mouseDown` and 'onClick' events should really be renamed if we are going with this solution, but that would presumably break a lot of stuff!

Reply:
Yup, but I mean more fundamentally... they are not the same right? It's surprising that to bind a callback to a `mousedown` event for it to be called on `mouseup`.

Reply:
That's a good point, maybe they sould have abstract names in week and above?

Reply:
Something like `onDateFocused` and `onDateSelected`???

Reply:
Yeah something to that effect